### PR TITLE
Some fixes for the pagination in the `Table` component.

### PR DESCRIPTION
**Pull request by cybervinvin**
## Summary

The goal for this PR is to fix some details on the `Table` component (especially on `Pagination`) 

(Internal for Opendatasoft only) Associated Shortcut ticket:
[sc-48044](https://app.shortcut.com/opendatasoft/story/48044/sdk-add-direction-rtl-to-table)
[sc-51025](https://app.shortcut.com/opendatasoft/story/51025/update-table-pagination-style)
[sc-50986](https://app.shortcut.com/opendatasoft/story/50986/pagination-visually-breaks-when-page-number-is-above-100)

### Changes

- Adding the `storybook-addon-rtl` storybook addon
- Make the `Pagination` component compliant with RTL direction
- Fix the `Cell`component: don't force left alignment
- Fix the `Pagination` component: superposition of page numbers (when the number of pages is greater than 999)
- Fix the styles of `Pagination` to match design
- Fix the `Table` component: `records` can be undefined
- Add default background-color on `Table`
- Fix backgound-opacity on the select's arrows (`PageSize`)
- Rename `border-radius-x` variable to match the value

#### Breaking Changes

`records` property is no longer obligatory

### Changelog

Some fixes for the pagination in the `Table` component.

## Open discussion

always welcome

## Review checklist

- [X] Description is complete
- [X] Commits respect the [Conventional Commits Specification](https://github.com/opendatasoft/ods-dataviz-sdk/blob/main/CONTRIBUTING.md#commit-messages)
- [X] 2 reviewers (1 if trivial)
- [X] Tests coverage has improved
- [X] Code is ready for a release on NPM

### CODE REVIEW

File: packages/visualizations-react/.storybook/main.js [hunk 1]
```diff
@@ -6,7 +6,11 @@ function getAbsolutePath(value) {
 
 export default {
     stories: ['../stories/**/*.stories.@(ts|tsx|js|jsx)'],
-    addons: [getAbsolutePath("@storybook/addon-links"), getAbsolutePath("@storybook/addon-essentials")],
+    addons: [
+        getAbsolutePath("@storybook/addon-links"),
+        getAbsolutePath("@storybook/addon-essentials"),
+        "storybook-addon-rtl"
```
Comment (KevinFabre-ods) [inline packages/visualizations-react/.storybook/main.js:12]: I would add a RTL table story for Chromatic to ensure we don't break the RTL layout in future PRs

**Review by KevinFabre-ods — COMMENTED** [review]
Looking good, I left some comments as you are still working on it

File: packages/visualizations/src/components/Pagination/Pagination.svelte [hunk 2]
```diff
@@ -15,7 +15,7 @@
     $: totalPages = Math.ceil(totalRecords / recordsPerPage);
 </script>
 
-<div class="pagination-container">
+<div class="pagination-container" dir="ltr">
```
[the remark this answers is not in the document]
Comment (cybervinvin) [reply packages/visualizations/src/components/Pagination/Pagination.svelte]: done

File: packages/visualizations/src/components/Pagination/Pagination.svelte [hunk 3]
```diff
@@ -62,6 +62,10 @@
         }
     }
 
+    :global(.table-container .pagination) {
+        border-top: solid 1px var(--border-color);
+    }
+
```
[the remark this answers is not in the document]
Comment (cybervinvin) [reply packages/visualizations/src/components/Pagination/Pagination.svelte]: done

File: packages/visualizations-react/.storybook/main.js (hunk 1 above)
Comment (cybervinvin) [reply packages/visualizations-react/.storybook/main.js:12]: done

**Review by KevinFabre-ods — APPROVED** [review]
Looks good 🚀 

Just a comment about the RTL story
https://www.chromatic.com/test?appId=616594e84c1692003af430f3&id=66878fe6ed9a0a5ec66738d1
RTL direction is not applied for this story in Chromatic

We might want to hide from Chromatic if we don't find a solution

File: packages/visualizations/src/components/Pagination/Pagination.svelte [hunk 4]
```diff
@@ -58,7 +58,8 @@
         }
         .page-size {
             grid-area: size;
-            margin-left: auto;
+            display: flex;
+            flex-direction: row-reverse;
```
Comment (etienneburdet) [inline packages/visualizations/src/components/Pagination/Pagination.svelte:62]:
```suggestion
    justify-self: end;
```

File: packages/visualizations/src/components/Table/TableCard.svelte [hunk 5]
```diff
@@ -41,10 +41,20 @@
...
+        overflow: hidden;
+        background-color: var(--background-color);
+    }
     :global(.ods-dataviz--default) div {
         max-width: 100%;
         display: flex;
         flex-direction: column;
         justify-content: space-between;
     }
+    :global(.table-container .pagination) {
```
Comment (etienneburdet) [inline packages/visualizations/src/components/Table/TableCard.svelte:57]: Has to be done Pagination, else .pagination isn't scoped and risks concflicts.

**Review by etienneburdet — CHANGES_REQUESTED** [review]
Just a few small changes for scoping and css, but looking good overall.

File: package-lock.json [hunk 6]
```diff
@@ -25729,6 +25729,12 @@
       "dev": true,
       "license": "MIT"
     },
+    "node_modules/storybook-addon-rtl": {
```
[the remark this answers is not in the document]
Comment (KevinFabre-ods) [reply package-lock.json:25732]: I don't think so, this file is shared between all three packages

File: packages/visualizations/src/components/Table/TableCard.svelte (hunk 5 above)
Comment (KevinFabre-ods) [reply packages/visualizations/src/components/Table/TableCard.svelte:57]: I think we've made assumption on how Svelte scopes CSS

A class name is not transformed.

image

In this example, the HTML `h1` will have a class attribute with a value: `'title'` 
And the style is only applied for this component tag as Svelte transforms the `h1` selector into a scoped selector`.title.svelte-zxiuva`

So IMO my point (https://github.com/opendatasoft/ods-dataviz-sdk/pull/253#discussion_r1663707612) stands. 
We should do a full rework on classes we want to expose to end users

**Comment by KevinFabre-ods** [comment]
Our issue with global class names isn't limited to the Table component.
We'll handle this in a dedicated PR (https://app.shortcut.com/opendatasoft/story/50530/sdk-better-global-classname)

**Pull request merged**